src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmrc.jenkinsjobbuilders.domain.trigger
+
+import uk.gov.hmrc.jenkinsjobbuilders.domain.Condition
+
+class UpstreamTrigger implements Trigger {
+    String upstreamJobs
+    Condition condition
+
+    private UpstreamTrigger(String upstreamJobs, Condition condition) {
+        this.upstreamJobs = upstreamJobs
+        this.condition = condition
+    }
+
+    /**
+     * Creates a {@link jenkins.triggers.ReverseBuildTrigger} based on the successful completion of any of a list of upstream jobs.
+     * @param upstreamJobs String with a comma separated list of jobs
+     * @param {@link uk.gov.hmrc.jenkinsjobbuilders.domain.Condition} one of SUCCESS, UNSTABLE, FAILURE
+     * @return UpstreamTrigger
+     */
+    static UpstreamTrigger upstreamTrigger(String upstreamJobs, Condition condition) {
```

Comment:
Should users define `condidtion` always? They will probably usually want success anyway. Maybe `condiditon` could be defaulted on line 7 (`Condition condition = Condition.SUCCESS`) and we provide a `withCondition(Condidtion condition)` where people could override in rare cases?

Reply:
Thanks, we'll make those changes.

Reply:
Comments addressed by https://github.com/hmrc/jenkins-job-builders/pull/41/commits/3f156f857ce5ab2bcf0d82008ad4b04287805c9f